File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int help(int n,vector<int> &dp)
    {
        if(n<2) return n;
        if(dp[n]!=-1) return dp[n];

        return dp[n]=help(n-1,dp)+help(n-2,dp);
    }
    int fib(int n) {
        vector<int> dp(n+1,-1);
        return help(n,dp);
    }
};

File: 1013-fibonacci-number/1013-fibonacci-number.py
class Solution(object):
    l=[0]
    def help(self,n):
        if n<2:
            return n
        if(len(self.l)>=n):
            return self.l[n-1]
        self.l.append(self.help(n-1)+self.help(n-2))
        return self.l[n-1]
    def fib(self, n):
       return self.help(n)
        